staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/apiserver.go

```diff
@@ -203,7 +203,11 @@ func (c completedConfig) New(delegationTarget genericapiserver.DelegationTarget)
 		crdHandler,
 	)
 	var openapiController *openapicontroller.Controller
-	if utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CustomResourcePublishOpenAPI) {
+	// OpenAPIVersionedService and StaticOpenAPISpec together serve the /openapi/v2 endpoint on
+	// s.GenericAPIServer. When a generic apiserver is brought up, it may choose to not enable
+	// OpenAPI by having null openAPIConfig, and thus OpenAPIVersionedService and StaticOpenAPISpec
+	// are both null. In that case we don't run the CRD OpenAPI controller.
+	if utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CustomResourcePublishOpenAPI) && s.GenericAPIServer.OpenAPIVersionedService != nil && s.GenericAPIServer.StaticOpenAPISpec != nil {
```

Comment:
OpenAPIVersionedService and StaticOpenAPISpec are populated in PrepareRun(), right? I wouldn't guess they'd be populated here

Reply:
ah that's right. fixing